zio-http/src/main/scala/zio/http/HttpApp.scala

```diff
@@ -0,0 +1,21 @@
+package zio.http
+
+import zio._
+
+/**
+ * A [[zio.http.HttpApp]] is a ready-to-execute HTTP application, which handles
+ * all requests, and always produces valid responses, translating application
+ * errors into HTTP errors.
+ */
+final case class HttpApp2[-Env](
+  requestHandler: Handler[Env, Response, Request, Response],
+  fatalHandler: Cause[Any] => ZIO[Env, Option[Nothing], Response],
```

Comment:
We can mention in the doc comment that it is not always possible to use `fatalHandler`'s result value but it will still be called for its side effects

Reply:
Unfortunately I have to delete this because `fatalHandler` cannot compose. I think fatal errors must be handled at server level.